recipes/libressl/2.9.2/conanfile.py

```diff
@@ -0,0 +1,73 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class LibreSSLConan(ConanFile):
+    name = "libressl"
+    version = "2.9.2"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.libressl.org/"
+    license = "OpenSSL - OpenBSD"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = "shared=False", "fPIC=True"
+    topics = ("SSL", "TLS", "openssl")
+    description = "LibreSSL is a version of the TLS/crypto stack forked from OpenSSL in 2014, with goals of modernizing the codebase, improving security, and applying best practice development processes."
+    generators = "cmake"
+
+    @property
+    def libressl_src(self):
+        return "%s-%s" % (self.name, self.version)
+
+    def config_options(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+
+    def build(self):
+        # needed to properly overwrite MSVC runtime
+        tools.replace_in_file(os.path.join(self.libressl_src, "CMakeLists.txt"), "project (LibreSSL C ASM)", """project (LibreSSL C ASM)
+include(${CMAKE_CURRENT_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)""")
+
+        # libressl adds a suffix to Windows dll names, which is really annoying for packaging
+        tools.replace_in_file(os.path.join(self.libressl_src, "crypto", "CMakeLists.txt"), "set(CRYPTO_POSTFIX -${CRYPTO_MAJOR_VERSION})", "set(CRYPTO_POSTFIX)")
+        tools.replace_in_file(os.path.join(self.libressl_src, "tls", "CMakeLists.txt"), "set(TLS_POSTFIX -${TLS_MAJOR_VERSION})", "set(TLS_POSTFIX)")
+        tools.replace_in_file(os.path.join(self.libressl_src, "ssl", "CMakeLists.txt"), "set(SSL_POSTFIX -${SSL_MAJOR_VERSION})", "set(SSL_POSTFIX)")
+        cmake = CMake(self)
+        cmake.definitions["BUILD_SHARED"] = self.options.shared
+        cmake.definitions["USE_SHARED"] = self.options.shared
+        # LibreSSL's CMake install rules are a bit weird
+        # explicitly remove CMAKE_INSTALL_PREFIX to avoid installing certs
+        cmake.definitions["CMAKE_INSTALL_PREFIX"] = ""
```

Comment:
Needed?

Reply:
I'm not sure about it, we had a lot of issues because we have to embed our own certificates on some platforms.

Reply:
I'd love to have inputs from someone knowing that area very well (I'm not that someone)

Reply:
From the Conan side, `CMAKE_INSTALL_PREFIX` is adjusted to the package folder in case `cmake.install()` is issued for the package step, so I don't see it colliding as that is not used. Anyway, it will be great to have more info, but not a stopper from my side

Reply:
I remember why happens when you don't set it to empty string.

`CMAKE_INSTALL_PREFIX` is set by the CMake helper to the package folder.

LibreSSL uses this prefix as a replacement for `OPENSSLDIR` if the latter is not set.

Thus, your `OPENSSLDIR` gets assigned with the **absolute** path to your local cache. Running `strings` on the built libs:

>OPENSSLDIR: "/home/theo/.conan/data/LibreSSL/2.9.2/theo/bug/package/7bebe6d67e4cb2a292d8af84bff08c50d71c8729/etc/ssl"

Since certificates are not provided, and thus not installed, by LibreSSL, TLS handshakes errors (amongst others) will arise when running any executable.

Reply:
I remember what happens when you don't override CMAKE_INSTALL_PREFIX.
LibreSSL uses it as a replacement for OPENSSLDIR, if the latter is not set.

Since Conan adjusts it to the package folder, running `strings` on the built libs give the following:

>OPENSSLDIR: "/home/theo/.conan/data/LibreSSL/2.9.2/theo/bug/package/7bebe6d67e4cb2a292d8af84bff08c50d71c8729/etc/ssl"

Not what you want... As any executable using those libs will try to find certificates in a folder that only exists on my machine...

That's why we must set CMAKE_INSTALL_PREFIX to an empty string, and package the lib by hand

Reply:
Thanks for clarifying. I find this mechanism weird... Could the recipe set `OPENSSLDIR` instead then?

Reply:
It is quite weird indeed. I guess we can add an option `with_openssldir` that can take `ANY`. The default value would be `/etc/ssl`?

Also, LibreSSL considers `OPENSSLDIR` as not set if it is an empty string. So there is quite a bit of gymnastics to be done.

Reply:
Well, let's keep it as it is and we could improve it in the future if anyone reports an issue related to this. Thanks a lot for the explanation